src/components/Tags.tsx

```diff
@@ -20,122 +25,114 @@ const KeyCodes = {
   enter: 13,
 };
 
-type tagstruct = {
-  rank: number;
-  segment: string;
-  tag: { name: string };
-  _id: string;
-};
-
-interface MyProps {
-  tags: { id: string; text: string }[];
-  seg: any;
-  rank: number;
-  forceUpdate: any;
-}
-
-interface MyState {
-  tags: any;
-  suggestions: any;
-  k: number;
-}
-
 const delimiters = [KeyCodes.comma, KeyCodes.enter];
 
-class Tags extends React.Component<MyProps, MyState> {
-  constructor(props: MyProps) {
-    super(props);
-    this.state = {
-      tags: props.tags,
-      suggestions: suggestions,
-      k: 1,
+const Tags = ({
+  disabled,
+  currentSegment,
+  segmentIndex,
+  rank,
+  updateSegmentAt,
+  refresh,
+}: {
+  disabled: boolean;
+  currentSegment: Segment;
+  segmentIndex: number;
+  rank: number;
+  updateSegmentAt: (index: number, data: Partial<Segment>) => void;
+  refresh: [boolean];
+}): JSX.Element => {
+  const [reactTags, setReactTags] = React.useState([] as ReactTag[]);
```

Comment:
The react hook `useState` provides a nice and concise way of defining a state variable and its setter function.

Reply:
Very nice.